lib/Horde/Imap/Client/Ids.php

```diff
@@ -497,4 +497,25 @@ public function unserialize($data)
         }
     }
 
+    /**
+     * @return array
+     */
+    public function __serialize()
+    {
+        return array(
+            'd' => $this->duplicates,
+            's' => $this->_sequence,
+            'is' => $this->_sorted,
+            'ids' => $this->_ids,
+        );
+    }
+
+    public function __unserialize(array $data)
+    {
+        $this->duplicates = $data['d'];
+        $this->_sequence = $data['s'];
+        $this->_sorted = $data['is'];
+        $this->_ids = $data['ids'];
```

Comment:
how come these two functions don't implement all the same logic as the original methods?

Reply:
I tried to be clever bug I guess it's better if I deduplicate the two serialization methods and use the old logic